algorithms.md

```diff
@@ -26,7 +26,7 @@ code here
 ```python
 # general language-independent implementation
 def find_substr(string, substr):
-    for i in range(0, len(string)):
+    for i in range(0, len(string) - len(substring) + 1):
```

Comment:
It looks like if we have two strings of lenght zero - why do we need to add this extra one mumber to have an empty iteration over empty strings?